### PR TITLE
CLOUD-833: Replace todos and examples in our Minion configuration schema

### PR DESCRIPTION
**Pull request by indigo423**
### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: the subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue that points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [x] If this is a new or updated feature, is there documentation for the new behavior?
* [x] If this is new code, are there unit and/or integration tests?
* [x] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Reviewer Hints

Adding reasonable descriptions and examples to our Minion configuration schema.

* Configuration for `/ipc/grpc/host-with-port` is split into `/ipc/grpc/host` and `/ipc/grpc/port`

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/CLOUD-833

### CODE REVIEW

File: opennms-container/minion/minion-config-schema.yml [hunk 1]
```diff
@@ -201,58 +201,65 @@ categories:
...
           - name: "host"
             type:
-              name: "host-with-port"
+              name: "host"
             label: "Host"
-            description: "TODO"
+            description: "gRPC host endpoint, typcally the same Core instance"
             key: "/ipc/grpc/host"
-            example: "TODO"
+            example: "my-opennms-instance-ip"
```
Comment (indigo423) [inline opennms-container/minion/minion-config-schema.yml:248]: @mattixtech the original was host-with-port. In our [docs](https://docs.opennms.com/horizon/28.0.0-SNAPSHOT/deployment/minion/install.html) I've found it with host and port separate. That's the reason I've added the additional config port section here. Does this make sense or should I revert it?

Screenshot 2021-01-21 at 10 48 48

File: opennms-container/minion/minion-config-schema.yml [hunk 2]
```diff
@@ -38,54 +38,54 @@ categories:
...
+            description: "A unique Minion identifier in the location."
             key: "/id"
-            example: "TODO"
+            example: "id: my-minion-nyc"
           - name: "broker-url"
             type:
               name: "url"
-            label: "Broker URL"
-            description: "TODO"
+            label: "The message broker URL"
```
Comment (bouff) [inline opennms-container/minion/minion-config-schema.yml:69]: Perhaps `Message broker URL` for brevity

File: opennms-container/minion/minion-config-schema.yml [hunk 3]
```diff
@@ -201,58 +201,65 @@ categories:
...
-        description: "TODO"
+        description: "gRPC for IPC configuration"
         config:
           - name: "host"
             type:
-              name: "host-with-port"
+              name: "host"
             label: "Host"
-            description: "TODO"
+            description: "gRPC host endpoint, typcally the same Core instance"
```
Comment (bouff) [inline opennms-container/minion/minion-config-schema.yml:246]: Typo.  Is Core meant to be capitalized?

File: opennms-container/minion/minion-config-schema.yml [hunk 4]
```diff
@@ -123,74 +123,74 @@ categories:
...
+                description: "RMI registry host IP address"
                 key: "/karaf/management/rmi/registry/host"
-                example: "TODO"
+                example: "127.0.0.1"
               - name: "registry-port"
                 type:
                   name: "port"
                 label: "Registry Port"
-                description: "TODO"
+                description: "RMI listening TCP port"
```
Comment (bouff) [inline opennms-container/minion/minion-config-schema.yml:149]: Some descriptions for ports have "listening" mentioned. Should this be removed throughout for consistency?  Conversely, could add "listening" to the others?

File: opennms-container/minion/minion-config-schema.yml (hunk 1 above)
Comment (mattixtech) [reply opennms-container/minion/minion-config-schema.yml:248]: Yep, that looks better. We drop each of these values into the same cfg file written to by the example you showed so if the example is the right way to do it, your updates here reflect that.

**Review by mattixtech — APPROVED** [review]

**Comment by mattixtech** [comment]
Should we target the release-27 branch? I believe that is the version of Horizon/Minion we will be supporting in the short term for MaaS.

File: opennms-container/minion/minion-config-schema.yml (hunk 3 above)
Comment (Bonrob2) [reply opennms-container/minion/minion-config-schema.yml:246]: I agree core should be lower case.

File: opennms-container/minion/minion-config-schema.yml [hunk 5]
```diff
@@ -12,23 +12,23 @@ categories:
...
-    description: "TODO"
+    description: "General Minion settings."
     config:
       - name: "instance-id"
         type:
           name: "string"
           validation: "TODO"
-        label: "OpenNMS Instance Id"
-        description: "TODO"
+        label: "Core Instance Id"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:21]:
```suggestion
        label: "Core instance ID"
```

File: opennms-container/minion/minion-config-schema.yml [hunk 6]
```diff
@@ -12,23 +12,23 @@ categories:
...
+        description: "Identifier for the Core instance used as queue prefix in the message broker."
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:22]:
```suggestion
        description: "Identifier for the core instance used as queue prefix in the message broker."
```

File: opennms-container/minion/minion-config-schema.yml [hunk 7]
```diff
@@ -12,23 +12,23 @@ categories:
...
         key: "/org.opennms.instance.id"
-        example: "TODO"
+        example: "OpenNMS"
       - name: "system-properties"
         type:
           name: "properties"
         label: "System Properties"
-        description: "TODO"
+        description: "Set generic Java system properties"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:29]:
```suggestion
        description: "Set generic Java system properties."
```

File: opennms-container/minion/minion-config-schema.yml [hunk 8]
```diff
@@ -38,54 +38,54 @@ categories:
...
+        label: "Minion Controller"
+        description: "Minion controller configuration for a location and communication endpoints."
         config:
           - name: "location"
             type:
               name: "string"
               validation: "TODO"
             label: "Location"
-            description: "TODO"
+            description: "Name of the location the Minion is deployed"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:55]:
```suggestion
            description: "Name of the location the Minion is deployed."
```

File: opennms-container/minion/minion-config-schema.yml [hunk 9]
```diff
@@ -38,54 +38,54 @@ categories:
...
             key: "/broker-url"
-            example: "TODO"
+            example: "ActiveMQ: failover:[tcp,ssl]://my-activemq-ip:61616, Kafka: broker-01:9092, broker-02:9092"
           - name: "http-url"
             type:
               name: "url"
-            label: "HTTP URL"
-            description: "TODO"
+            label: "Core base URL"
+            description: "The Core web application base URL."
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:77]:
```suggestion
            description: "Core web application base URL"
```

File: opennms-container/minion/minion-config-schema.yml (hunk 2 above)
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:69]:
```suggestion
            label: "Message broker URL"
```

File: opennms-container/minion/minion-config-schema.yml [hunk 10]
```diff
@@ -123,74 +123,74 @@ categories:
...
+                description: "RMI listening TCP port"
                 key: "/karaf/management/rmi/registry/port"
-                example: "TODO"
+                example: "1299"
               - name: "server-host"
                 type:
                   name: "host"
                 label: "Server Host"
-                description: "TODO"
+                description: "RMI server host IP address seen by the client"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:156]:
```suggestion
                description: "RMI server host IP address seen by the client."
```

File: opennms-container/minion/minion-config-schema.yml [hunk 11]
```diff
@@ -123,74 +123,74 @@ categories:
...
-                description: "TODO"
+                description: "RMI server TCP port"
                 key: "/karaf/management/rmi/server/port"
-                example: "TODO"
+                example: "45444"
       # Karaf :: Shell
       - name: "shell"
         label: "Shell"
-        description: "TODO"
+        description: "Karaf Shell access configuration"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:169]:
```suggestion
        description: "Karaf shell access configuration"
```

File: opennms-container/minion/minion-config-schema.yml [hunk 12]
```diff
@@ -123,74 +123,74 @@ categories:
...
+                description: "Listening IP address for the SSH service"
                 key: "/karaf/shell/ssh"
-                example: "TODO"
+                example: "0.0.0.0"
               - name: "port"
                 type:
                   name: "port"
                 label: "Port"
-                description: "TODO"
+                description: "TCP listening port the SSH service"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:187]:
```suggestion
                description: "TCP listening port for the SSH service"
```

File: opennms-container/minion/minion-config-schema.yml [hunk 13]
```diff
@@ -201,58 +201,65 @@ categories:
...
-        description: "TODO"
+        description: "Additional Java options"
         key: "/process-env/java-opts"
-        example: "TODO"
+        example: "-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=7896"
   # IPC Configuration
   - name: "ipc"
     label: "IPC"
-    description: "TODO"
+    description: "Core, Minion and Sentinel distributed inter process communication (IPC) settings"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:210]:
```suggestion
    description: "Core, Minion, and Sentinel distributed inter-process communication (IPC) settings"
```

File: opennms-container/minion/minion-config-schema.yml [hunk 14]
```diff
@@ -201,58 +201,65 @@ categories:
...
-            description: "TODO"
+            description: "gRPC host endpoint, typcally the same Core instance"
             key: "/ipc/grpc/host"
-            example: "TODO"
+            example: "my-opennms-instance-ip"
+          - name: "port"
+            type:
+              name: "port"
+            label: "gRPC TCP port"
+            description: "gRPC TCP port the server is listening"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:253]:
```suggestion
            description: "gRPC TCP port the server is listening on."
```

File: opennms-container/minion/minion-config-schema.yml [hunk 15]
```diff
@@ -12,23 +12,23 @@ categories:
   # General
   - name: "general"
     label: "General"
-    description: "TODO"
+    description: "General Minion settings."
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:15]:
```suggestion
    description: "General Minion settings"
```

File: opennms-container/minion/minion-config-schema.yml [hunk 16]
```diff
@@ -123,74 +123,74 @@ categories:
...
       - name: "management"
         label: "Management"
-        description: "TODO"
+        description: "Configuration for Karaf JMX management access"
         categories:
           # Karaf :: Management :: RMI
           - name: "rmi"
             label: "RMI"
-            description: "TODO"
+            description: "Configure remote method invocation (RMI)"
```
Comment (Bonrob2) [inline opennms-container/minion/minion-config-schema.yml:136]: Not a big deal, but this one is more of a command. Should it be "Configuration for remote method invocation (RMI)" like line 131?

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
a few minor changes.

File: opennms-container/minion/minion-config-schema.yml [hunk 17]
```diff
@@ -201,67 +201,74 @@ categories:
...
-        description: "TODO"
+        description: "gRPC for IPC configuration"
         config:
           - name: "host"
             type:
-              name: "host-with-port"
+              name: "host"
             label: "Host"
-            description: "TODO"
+            description: "gRPC host endpoint, typcally the same as the core instance"
```
Comment (bouff) [inline opennms-container/minion/minion-config-schema.yml:246]: Typo

File: opennms-container/minion/minion-config-schema.yml [hunk 18]
```diff
@@ -435,41 +442,41 @@ categories:
...
             label: "Single Port Listener"
-            description: "TODO"
+            description: "Configuration for single port flow listener"
             config:
               - name: "enabled"
                 type:
                   name: "boolean"
                 label: "Enable Single Port Listener"
-                description: "TODO"
+                description: "Enable or disable the single port flow listener."
```
Comment (bouff) [inline opennms-container/minion/minion-config-schema.yml:458]: Note on consistency, most descriptions don't end with a `.` - but there are a few exceptions here and there

**Review by bouff — APPROVED** [review]

File: opennms-container/minion/minion-config-schema.yml (hunk 18 above)
Comment (indigo423) [reply opennms-container/minion/minion-config-schema.yml:458]: @bouff thanks, I agree and addressed this topic. Thank you very much!

**Review by mattixtech — APPROVED** [review]

**Pull request merged**